src/ZoneServer/Buffs/Handlers/Hold.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using Melia.Shared.Game.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Network;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using Yggdrasil.Logging;
+
+namespace Melia.Zone.Buffs.Handlers
+{
+	/// <summary>
+	/// Handle for the Hold Debuff, which prevents movement
+	/// </summary>
+	[BuffHandler(BuffId.Hold)]
+	public class Hold : BuffHandler
+	{
+		public override void OnStart(Buff buff)
+		{
+			var target = buff.Target;
+
+			if (target.Components.TryGet<MovementComponent>(out var movementComponent))
+			{
+				movementComponent.Stop();
+				movementComponent.SetFixedMoveSpeed(0.0001f);
```

Comment:
I actually just based it on TheReturn's implementation of Leg Trap, since the debuffs have the same effect.  I agree this implementation seems like it could be flawed, as, for example, if you were to be placed under two sources of immobility, when one wore off, you would recover from both of them (since the End function would set your movement speed back to normal).  It's possible there may need to be some kind of flag for this.

Reply:
I suppose it's acceptable as a temporary implementation for now, just to have something, though I'm curious if tested what happens if you set the movement speed to 0, as in no movement at all, which seems like it would make more sense to me.

Reply:
Strangely, 0 does nothing (the monster's movement speed doesn't change at all).  With 0.0001 they seem like they are actually stopped (maybe it rounds to an int?), even when it was initially bugged and lasted forever they never actually moved anywhere even after like a full minute.

Reply:
I guess I'll have to look into this some time, because I just checked our movement code and it is set to do absolutely nothing on move if the speed is 0. Though that's an issue for another day.

Reply:
Ohhh, I figured out why 0 doesn't work. That's the default value returned if you request a property value that is not set, and we check the result against 0 to determine if it was set. So the monster movement speed property calculator function doesn't consider the 0 value valid and skips over it.